sdk/communication/communication-rooms/test/public/roomsClient.spec.ts

```diff
@@ -159,139 +134,108 @@ describe("RoomsClient", function () {
       testUser = (await createTestUser(recorder)).user;
-      const participants: RoomParticipant[] = [
+      const participants: RoomParticipantPatch[] = [
         {
           id: testUser,
           role: "Presenter",
         },
       ];
 
-      const createRoomResult = await client.createRoom({});
-      assert.isDefined(createRoomResult);
-      assert.isEmpty(createRoomResult.participants);
-      roomId = createRoomResult.id;
+      await client.addOrUpdateParticipants(roomId, participants);
+      await pause(500);
 
-      await client.addParticipants(roomId, participants);
-      const addParticipantsResult = await client.getParticipants(roomId);
+      const addParticipantsResult = await client.listParticipants(roomId);
       assert.isDefined(addParticipantsResult);
       assert.isNotEmpty(addParticipantsResult);
-      assert.equal(
-        getIdentifierRawId(addParticipantsResult[0].id),
-        getIdentifierRawId(participants[0].id)
-      );
-      assert.equal(addParticipantsResult[0].role, participants[0].role);
-    });
 
-    it("successfully removes a participant from the room", async function () {
-      testUser = (await createTestUser(recorder)).user;
-      testUser2 = (await createTestUser(recorder)).user;
+      let count = 0;
+      for await (const participant of addParticipantsResult) {
+        if (participant) {
+          count++;
 
-      const request: CreateRoomOptions = {
-        participants: [
-          {
-            id: testUser,
-            role: "Presenter",
-          },
-          {
-            id: testUser2,
-            role: "Presenter",
-          },
-        ],
-      };
-
-      const createRoomResult = await client.createRoom(request);
-      assert.isDefined(createRoomResult);
-      assert.isNotEmpty(createRoomResult.participants);
-
-      roomId = createRoomResult.id;
+          // rawId is sanitized so skip this check in playback mode
+          if (!isPlaybackMode()) {
+            assert.equal(participant.id.kind, "communicationUser");
+          }
+        }
+      }
 
-      const participants = [testUser, testUser2];
-      await client.removeParticipants(roomId, participants);
-      assert.isEmpty((await client.getRoom(roomId)).participants);
+      assert.equal(count, 1);
     });
 
     it("successfully updates a participant", async function () {
-      testUser = (await createTestUser(recorder)).user;
-      const request: CreateRoomOptions = {
-        participants: [
-          {
-            id: testUser,
-            role: "Presenter",
-          },
-        ],
-      };
-
-      const createRoomResult = await client.createRoom(request);
-      assert.isDefined(createRoomResult);
-      assert.isNotEmpty(createRoomResult.participants);
-      roomId = createRoomResult.id;
-
-      const participants: RoomParticipant[] = [
+      const participants: RoomParticipantPatch[] = [
         {
           id: testUser,
```

Comment:
Do we have a test case for when role is explicitly set to null? This should still default to Attendee role
```
const participants: RoomParticipantPatch[] = [
{
  "id": testUser,
   "role": null 
}
]
```